airflow/migrations/versions/d38e04c12aa2_add_serialized_dag_table.py

```diff
@@ -0,0 +1,50 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+# 
+#   http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""add serialized_dag table
+
+Revision ID: d38e04c12aa2
+Revises: 6e96a59344a4
+Create Date: 2019-08-01 14:39:35.616417
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = 'd38e04c12aa2'
+down_revision = '6e96a59344a4'
+branch_labels = None
+depends_on = None
+
+from alembic import op
+import sqlalchemy as sa
+
+
+def upgrade():
+    op.create_table('serialized_dag',
```

Comment:
Any reason why we need a new table instead of extending the existing `dag` table?

Reply:
Thanks for asking. There are a few benefits to create a few table:

- As this feature is new and may be unstable, we would like to minimize its influence on current Airflow code (and db performance), especially things on the critical path. Using a separate table is safer. At least we know the dag table's behavior is not changed
- Merging these two tables make the dag table wide: it needs a hash of file path, a 'DELETED' maker, and a JSON data column.
- Logically serialized_dag is a snapshot of the files we have in DAG folder, but the dag table is not

I am open for changes. Any ideas?

Reply:
And as we want this feature to be completely optional till 2.0 - it is better to have a new table instead.

Reply:
Make sense